crates/swc_ecma_minifier/src/metadata/mod.rs

```diff
@@ -167,10 +180,12 @@ impl VisitMut for InfoMarker<'_> {
 
         if self.has_pure(n.span) {
             n.span = n.span.apply_mark(self.marks.pure);
-        } else if let Some(pure_fns) = &self.pure_funcs {
+        }
+
+        if !self.pure_funcs.is_empty() {
             if let Callee::Expr(e) = &n.callee {
                 // Check for pure_funcs
-                if pure_fns.contains(&HashEqIgnoreSpanExprRef(e)) {
+                if self.pure_funcs.contains(&HashEqIgnoreSpanExprRef(e)) {
```

Comment:
Can we use two hashset instead?
I think it will avoid allocations

Reply:
One hashset with type `Lazy<FxHashSet<HashEqIgnoreSpanExprRef<'static>>>`

Reply:
Get it.